### PR TITLE
neovim: update to 0.4.2.

### PR DESCRIPTION
**Pull request by AluisioASG**
Basic editing works, didn't try a GUI or using the Lua API.

Didn't see #13648 while working on this, might be worth waiting for that.

### CODE REVIEW

**Comment by Chocimier** [comment]
libluv merged

**Comment by AluisioASG** [comment]
PR updated and tested without libluv commit.

File: srcpkgs/libvterm/template [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Template file for 'libvterm'
 pkgname=libvterm
-version=0.0.20151229
-revision=2
-_commit=04c0777b139cfbddb057c0cbfc007677dd9f1b4f
+version=0.1
+revision=1
+_commit=f8ab54cf592b945caaaf2bc397c0da6fc8fb949b
```
Comment (Chocimier) [inline srcpkgs/libvterm/template]: Original 0.1 https://launchpad.net/libvterm is tagged, can we switch?

Comment (AluisioASG) [reply srcpkgs/libvterm/template]: I'll try it in a moment. Here's [the diff](https://gist.github.com/AluisioASG/87ca7d13fc48593b2b754e93353b2d3f) between upstream 0.1.1 and the nvim branch, for reference.

**Comment by Chocimier** [comment]
neovim now requires `libvterm.so.0.0.1`, not `.0.0.0` provided by earlier version of package, but xbps-src checks for `libvterm.so.0`, what allows to update neovim without update of libvterm.

Not sure if bumping shlibs to `libvterm.so.0 libvterm-0.0.1_1` correct solution.

File: srcpkgs/neovim/template [hunk 2]
```diff
@@ -1,17 +1,18 @@
 # Template file for 'neovim'
 pkgname=neovim
-version=0.3.8
+version=0.4.2
 revision=1
 build_style=cmake
 configure_args="-DENABLE_JEMALLOC=0"
```
Comment (concatime) [inline srcpkgs/neovim/template]: Remove this line, jemalloc is not used anymore https://github.com/neovim/neovim/pull/9526.

**Comment by AluisioASG** [comment]
@Chocimier I went with specifying the version manually in `depends` for now. `shlibs` can be updated later if someone thinks it best.

**Comment by gbrlsnchs** [comment]
Is something blocking merging this?

**Comment by Hoshpak** [comment]
Looks good to me.

**Pull request merged**